internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -29,13 +30,16 @@ var (
 )
 
 // GetTlsConfigurationForClient returns a tls.Config instance for use by a client application.
-func GetTlsConfigurationForClient(serverHostname string, cstream s2av2pb.S2AService_SetUpSessionClient) (*tls.Config, error) {
+func GetTlsConfigurationForClient(serverHostname string, cstream s2av2pb.S2AService_SetUpSessionClient, tokenManager tokenmanager.AccessTokenManager) (*tls.Config, error) {
+	token, err := tokenManager.DefaultToken()
```

Comment:
This logic needs to be a little different: we need to accept a local identity pointer to GetTlsConfigurationForClient as well, if this pointer is non-nil then we call the tokenManager's Token(local identity) API, and if this pointer is nil then we call tokenManager.DefaultToken.